File: Easy/C/The Largest Gap.c
#include <inttypes.h>
#include <stdio.h>

int main() {
    uint16_t N;
    scanf("%" SCNu16, &N);
    // Even though the problem states that the maximum value for any element is 10000, using uint16_t fails for TestCase#1 and TestCase#2
    uint32_t biggest = 0, smallest = UINT32_MAX;
    for (uint16_t i = 0; i < N; ++i) {
        uint32_t element;
        scanf("%" SCNu32, &element);
        if (element > biggest)
            biggest = element;

        if (element < smallest)
            smallest = element;
    }
    printf("%" PRIu32, biggest - smallest);
}

File: Easy/Rust/The Largest Gap.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    let N: u16 = lines.next().unwrap().unwrap().trim().parse().unwrap();
    // Even though the problem states that the maximum value for any element is 10000, using u16 fails for TestCase#1 and TestCase#2
    let Array: Vec<u32> = lines
        .next()
        .unwrap()
        .unwrap()
        .split_whitespace()
        .take(N as usize)
        .map(|element| element.parse().unwrap())
        .collect();
    print!(
        "{}",
        Array.iter().max().unwrap() - Array.iter().min().unwrap()
    );
}
